docs/threadpools.md

```diff
@@ -0,0 +1,34 @@
+Starlette supports both `async` and `sync` routes and background tasks.
+The sync part is running inside a threadpool by using [`AnyIO`](https://github.com/agronholm/anyio).
+
+The `AnyIO` library by default starts a threadpool with a fixed number of threads, which works for most cases.
+But since the threadpool is shared by both routes and background tasks,
+it might be useful to control the number of threads.
+
+
+```py
+import anyio
+from starlette.applications import Starlette
+from starlette.background import BackgroundTasks
+from starlette.request import Request
+from starlette.responses import JSONResponse
+from starlette.routing import Route
+
+# Set number of threads in threadpool to 100
+anyio.to_thread.current_default_thread_limiter().total_tokens = 100
```

Comment:
```suggestion
# Set number of threads in threadpool to 100
# Note that this increases the limit for anything using AnyIO, not just Starlette
anyio.to_thread.current_default_thread_limiter().total_tokens = 100
```

Reply:
Devil's advocate: if we document this would it then be breaking change if we start using our own capacity limiter(s), meaning this stops working? I think not, it's okay to break docs to some extent.

Reply:
We can add update the docs when we add the CapacityLimiter to explain this.